### DISCUSSION TITLE
Beatmap difficulty icon tooltip and colour are inaccurate

### DISCUSSION THREAD

**Discussion by Sinamer** [ppy/osu#15525] [Q&A] [answered]
Hovering over the difficulty icon shows the star rating of the beatmap, but it's not the actual star rating:

https://user-images.githubusercontent.com/53346335/140747201-aa90522d-cc45-4613-ad4d-a59bfba0ca4a.mp4

**Comment by peppy** [comment, marked as answer]
Are you running on master or something? If so then star ratings have changes and this is expected

**Reply by Sinamer** [reply]
Nope, this is the 2021.1105.0-lazer release from 4 days ago.

**Reply by peppy** [reply]
It's probably a similar thing if you imported beatmaps a while back.

**Reply by Sinamer** [reply]
Right, it works as expected on the beatmaps that were just downloaded. Any way to force a recalculation so the star rating is in sync?

**Reply by bdach** [reply]
unfortunately no. this is similar to https://github.com/ppy/osu/issues/12048 in that the difficulty is only computed once on import, but the resolution would need to come from a slightly different angle.

we probably would need to add some sort of versioning to the diffcalc algorithm and force a recalc when that version changes.

**Reply by smoogipoo** [reply]
Will happen after the realm database migration is completed.
